File: cpp/KimElektro.cpp
#include <bits/stdc++.h>  // Mengimpor hampir semua header standar C++ sekaligus
using namespace std;      // Menggunakan namespace std agar tidak perlu menulis std:: sebelum fungsi/kelas STL

// Kelas KimElektro untuk merepresentasikan produk elektronik
class KimElektro {
private:
    string id;        // ID unik setiap produk
    string nama;      // Nama produk
    string JenisAlat;  // JenisAlat produk
    int harga;        // Harga produk dalam bentuk integer

public:
    // Constructor: digunakan untuk membuat objek KimElektro baru
    KimElektro(string _id, string _nama, string _JenisAlat, int _harga)
        : id(_id), nama(_nama), JenisAlat(_JenisAlat), harga(_harga) {}

    // Getter: mengambil nilai atribut
    string getID() { return id; }
    string getNama() { return nama; }
    string getJenisAlat() { return JenisAlat; }
    int getHarga() { return harga; }

    // Setter: mengubah nilai atribut
    void setID(string _id) { id = _id; }
    void setNama(string _nama) { nama = _nama; }
    void setJenisAlat(string _JenisAlat) { JenisAlat = _JenisAlat; }
    void setHarga(int _harga) { harga = _harga; }

    // Method untuk mencetak info barang ke console
    void printElektro() {
        cout << "ID      : " << id << endl;
        cout << "Nama    : " << nama << endl;
        cout << "JenisAlat: " << JenisAlat << endl;
        cout << "Harga   : " << harga << endl;
    }

    // Destructor: dipanggil saat objek dihapus (tidak melakukan aksi khusus di sini)
    ~KimElektro(){}
};

File: php/KimElektro.php
<?php
// =====================
// Class KimElektro
// Representasi barang elektronik di toko
// =====================
class KimElektro {
    private $id;       // ID unik barang
    private $nama;     // Nama barang
    private $JenisAlat; // JenisAlat barang
    private $harga;    // Harga barang (integer)
    private $gambar;   // Path file gambar lokal

    // =====================
    // Konstruktor
    // =====================
    public function __construct($id, $nama, $JenisAlat, $harga, $gambar=""){
        $this->id = $id;
        $this->nama = $nama;
        $this->JenisAlat = $JenisAlat;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // =====================
    // Getter (ambil nilai properti)
    // =====================
    public function getID(){ return $this->id; }
    public function getNama(){ return $this->nama; }
    public function getJenisAlat(){ return $this->JenisAlat; }
    public function getHarga(){ return $this->harga; }
    public function getGambar(){ return $this->gambar; }

    // =====================
    // Setter (ubah nilai properti)
    // =====================
    public function setID($id){ $this->id = $id; }
    public function setNama($nama){ $this->nama = $nama; }
    public function setJenisAlat($JenisAlat){ $this->JenisAlat = $JenisAlat; }
    public function setHarga($harga){ $this->harga = $harga; }
    public function setGambar($gambar){ $this->gambar = $gambar; }
}
?>